README.md

```diff
@@ -1,3 +1,150 @@
 # Juju Dashboard Charm
 
-This repository contains charms to deploy the dashboard for [Juju](https://juju.is) and [JAAS](https://jaas.ai) into a Machine and Kubernetes environments.
+This repository contains charms that deploy the dashboard for [Juju](https://juju.is) and [JAAS](https://jaas.ai) into Machine and Kubernetes environments.
+
+# Building and Testing the Machine Charm
+
+Building the machine charm is fairly straightforward. You need a machine with Juju 3.0+ installed. The rest of the steps, in brief, are:
+
+```
+# From the root directory of this repo
+
+# Build the charm
+cd ./machine-charm
+charmcraft pack
+
+# Bootstrap a test controller. Make a make a model, and deploy an application, so that the dashboard has things to show.
+juju bootstrap localhost localhost-test
+juju add-model test
+juju switch test
+juju deploy ubuntu
+
+# Switch to the controller model and setup the dashboard
+juju switch controller
+juju deploy ./juju-dashboard*.charm dashboard
+juju relate controller dashboard
+juju dashboard
+```
+
+Once you login following the instructions in the output of `juju dashboard` you should be able to see the controller model, as well as the test model and test application that you deployed.
+
+# Building and Testing the k8s charm
+
+This guide covers every step necessary to build and test a k8s charm, from updating the facades in js-libjuju, to rebuilding the dashboard source, to building the charm. You can skip some steps if you don't need to update every part of the toolchain.
+
+## Install Juju 3.0
+
+Bulid from [source](https://github.com/juju/juju), or `snap install --channel=3.0/beta juju`
+
+## Install Docker Engine
+
+Follow the [guide on the docker website](https://docs.docker.com/desktop/install/linux-install/) to install the latest version of the Docker engine.
+
+## Install nodejs and yarn
+
+TODO
+
+## Bootstrap microk8s
+[Follow the guide here](https://juju.is/docs/olm/microk8s)
+
+In brief:
+
+```
+snap install microk8s --classic
```

Comment:
The strict microk8s snap appears to have been released to `1.25-strict/stable`.
I am not exactly sure of the deprecation schedule for the classic snap but I think we should be transitioning to strict. Juju 3 integrates with the strict version and we will drop support for microk8s classic at some point I expect. With strict, I am not sure that you still need to be in the microk8s group anymore, but that needs fact checking.